### PR TITLE
Make save_main_session optional

### PR DESCRIPTION
**Pull request by silviulica**
This PR does 3 things:
1) make --save_main_session optional
2) Sets it to true in all examples that need it (e.g., wordcount needs it)
3) refactor setup.py to separate as much as possible setup logic from strings/versions.

### CODE REVIEW

**Comment by silviulica** [comment]
R: @robertwb please take a look

File: sdks/python/apache_beam/utils/dependency_test.py [hunk 1]
```diff
@@ -88,11 +88,8 @@ def test_default_resources(self):
     self.update_options(options)
 
     self.assertEqual(
```
Comment (robertwb) [inline sdks/python/apache_beam/utils/dependency_test.py]: It seems we should have at least one test with a main session.

File: sdks/python/apache_beam/examples/complete/autocomplete.py [hunk 2]
```diff
@@ -36,6 +36,9 @@ def run(argv=None):
                       required=True,
                       help='Output file to write results to.')
   known_args, pipeline_args = parser.parse_known_args(argv)
+  # We use the save_main_session option because one or more DoFn's in this
+  # workflow rely on global context (e.g., a module imported at module level).
```
Comment (robertwb) [inline sdks/python/apache_beam/examples/complete/autocomplete.py]: s/a module imported at module level/a module imported in __main__/g

File: sdks/python/apache_beam/examples/complete/estimate_pi.py [hunk 3]
```diff
@@ -106,6 +106,9 @@ def run(argv=None):
                       required=True,
                       help='Output file to write results to.')
   known_args, pipeline_args = parser.parse_known_args(argv)
+  # We use the save_main_session option because one or more DoFn's in this
+  # workflow rely on global context (e.g., a module imported at module level).
+  pipeline_args.append('--save-main-session')
```
Comment (robertwb) [inline sdks/python/apache_beam/examples/complete/estimate_pi.py]: I'd rather encourage the patter of creating an PipelineOptions and programatically setting this value rather than manipulating the flat strings.

File: sdks/python/setup.py [hunk 4]
```diff
@@ -15,13 +15,32 @@
 # limitations under the License.
 #
 
-"""Apache Beam SDK setup configuration."""
```
Comment (robertwb) [inline sdks/python/setup.py]: Seems these changes should be a separate commit.

**Comment by robertwb** [comment]
This'll be great!
